ntbs-service-unit-tests/Jobs/UpdateTableCountsJobTest.cs

```diff
@@ -0,0 +1,193 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Moq;
+using ntbs_service.DataAccess;
+using ntbs_service.Jobs;
+using ntbs_service.Models.QueryEntities;
+using Serilog;
+using Serilog.Events;
+using Serilog.Sinks.TestCorrelator;
+using Xunit;
+
+namespace ntbs_service_unit_tests.Jobs
+{
+    public class UpdateTableCountsJobTest
+    {
+        private readonly Mock<ITableCountsRepository> _tableCountsRepository;
+
+        private readonly UpdateTableCountsJob _updateTableCountsJob;
+
+        public UpdateTableCountsJobTest()
+        {
+            _tableCountsRepository = new Mock<ITableCountsRepository>();
+
+            _updateTableCountsJob = new UpdateTableCountsJob(_tableCountsRepository.Object);
+        }
+
+        [Fact]
+        public async Task Run_CallsUpdateStoredProcedure()
+        {
+            // Arrange
+            _tableCountsRepository
+                .Setup(r => r.GetRecentTableCounts())
+                .Returns(Task.FromResult(Enumerable.Repeat(CreateBasicTableCounts(), 2)));
+
+            // Act
+            await _updateTableCountsJob.Run(null);
+
+            // Assert
+            _tableCountsRepository.Verify(r => r.ExecuteUpdateTableCountsStoredProcedure());
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(1)]
```

Comment:
This is a slightly odd test because we're asserting that something will fail, even though in the real world it will pass (because the run will create another entry). I think it's ok, but probably worth adding a comment.

Reply:
This doesn't throw an exception any more as discussed above, I've rewritten this to check there is a message warning that there are not enough table counts available.